src/controllers/api/ruleSectionController.js

```diff
@@ -0,0 +1,35 @@
+const RuleSection = require('../../models/ruleSection');
+const utility = require('./utility');
+
+exports.index = async (req, res, next) => {
+  const search_queries = {};
+  if (req.query.name !== undefined) {
+    search_queries.name = { $regex: new RegExp(utility.escapeRegExp(req.query.name), 'i') };
+  }
+  if (req.query.desc !== undefined) {
+    search_queries.desc = { $regex: new RegExp(utility.escapeRegExp(req.query.desc), 'i') };
+  }
+
+  await RuleSection.find(search_queries)
```

Comment:
We should.... cache this. See Spells, Monsters, and Magic Items to see how I've handled this. You'll also want to look at them for tests too.